Linux/Car_Client/chronoscomm.cpp

```diff
@@ -806,30 +809,50 @@ bool ChronosComm::decodeMsg(quint16 type, quint32 len, QByteArray payload, uint8
     } break;
 
     case ISO_MSG_OSEM: {
+        qDebug() << "OSEM";
+
         chronos_osem osem;
 
         while (!vb.isEmpty()) {
             quint16 value_id = vb.vbPopFrontUint16();
             quint16 value_len = vb.vbPopFrontUint16();
+
             switch (value_id) {
-            case ISO_VALUE_ID_LAT:
+            case VALUE_ID_OSEM_TRANSMITTER_ID:
+                osem.transmitterID = vb.vbPopFrontUint32();
+                break;
+            case VALUE_ID_OSEM_LATITUDE :
                 osem.lat = vb.vbPopFrontDouble48(1e10);
                 break;
-            case ISO_VALUE_ID_LON:
+            case VALUE_ID_OSEM_LONGITUDE:
                 osem.lon = vb.vbPopFrontDouble48(1e10);
                 break;
-            case ISO_VALUE_ID_ALT:
+            case VALUE_ID_OSEM_ALTITUDE:
                 osem.alt = vb.vbPopFrontDouble32(1e2);
                 break;
-            case ISO_VALUE_ID_DateISO8601:
+            case VALUE_ID_OSEM_DATE:
                 vb.vbPopFrontUint32(); // pop and throw away
                 break;
-            case ISO_VALUE_ID_GPS_WEEK:
+            case VALUE_ID_OSEM_GPS_WEEK:
                 osem.gps_week = vb.vbPopFrontUint16();
                 break;
-            case ISO_VALUE_ID_GPS_SEC_OF_WEEK:
+            case VALUE_ID_OSEM_GPS_QUARTER_MILLISECOND_OF_WEEK:
                 osem.gps_ms_of_week = vb.vbPopFrontUint32() / 4;
                 break;
+                /*
```

Comment:
If you skip popping from the front when these values arrive, it becomes impossible to decode the rest of the message.

Reply:
Added the rest of the messages.